natural-language-understanding/v1.js

```diff
@@ -0,0 +1,184 @@
+  */
+QueryBuilder.prototype.getFeature = function(featureName, featureOptions) {
+  const options = featureOptions || {};
+  if (this.availableFeatures.has(featureName)) {
+    if (this._options.features) {
+      if (!this._options.features.hasOwnProperty(featureName)) {
+        this._options.features[featureName] = options;
+      }
+    } else {
+      this._options.features = {};
+      this._options.features[featureName] = options;
+    }
+    return this;
+  } else {
+    throw new Error(`${featureName} is not a valid feature`);
+  }
+}
+
+/**
+  * getFeatures:  builds a query adding the features to getFeature.
+  * @param {object} featureDict a dictionary of features and their options.
+  * @returns {self} returns the query builder so they can be chained.
+  */
+QueryBuilder.prototype.getFeatures = function(featureDict) {
+  Object.keys(featureDict).map((feature) => this.getFeature(feature, featureDict[feature]));
+  return this;
+}
+
+/**
+  * Get all available features
+  * @returns {self} returns the query builder so they can be chained
+  */
+QueryBuilder.prototype.getAllFeatures = function() {
+  return this.getFeatures(afeatures);
+}
+
+/**
+  * adds the HTML content to the query
+  * @param {string} htmlstring a string of html content
+  * @returns {self} returns the query builder so they can be chained
+  */
+QueryBuilder.prototype.withHtmlString = function(htmlstring) {
+  this._options.html = htmlstring;
+  if (this._options.text || this._options.url) {
+    throw new Error("Only one HTML, Text, or URL can be supplied");
+  }
+  return this;
+}
+
+/**
+  * adds the Text content to the query
+  * @param {string} textstring a string of text content
+  * @returns {self} returns the query builder so they can be chained
+  */
+QueryBuilder.prototype.withTextString = function(textstring) {
+  this._options.text = textstring;
+  if (this._options.html || this._options.url) {
+    throw new Error("Only one HTML, Text, or URL can be supplied");
+  }
+  return this;
+}
+
+/**
+  * adds the URL to fetch to the query
+  * @param {string} urlstring a string of a url to fetch
+  * @returns {self} returns the query builder so they can be chained
+  */
+QueryBuilder.prototype.withURL = function(urlstring) {
+  this._options.url = urlstring
+  if (this._options.html || this._options.text) {
+    throw new Error("Only one HTML, Text, or URL can be supplied");
+  }
+  return this;
+}
+
+/**
+  * Object returns the query data
+  * @returns {object} just the query options, formatted for the api
+  */
+QueryBuilder.prototype.object = function() {
+  return this._options;
+};
+
+/**
+  * Analyze the query.
+  * @params {QueryBuilder} query a QueryBuilder query
+  * @params {object} params any addtional request params, can be null
+  * @params {function} callback taking (error,  jsonResult)
+  * @returns {void}
+  */
+NaturalLanguageUnderstandingV1.prototype.analyze = function(query, params, callback) {
```

Comment:
I think we want to keep the signature to be `method(params, callback)`. It's a mental model at this point, having an extra parameter will make it difficult to differentiate when something is a query or when it's a param.